implementations/rust/ockam/ockam_core/src/bare.rs

```
@@ -0,0 +1,154 @@
+#![allow(missing_docs)]
+
+/// Defines a set of functions to encode and decode bare encoding
+///
+/// This module is not dependent on std or any other crate
+use crate::compat::vec::Vec;
+
+/// Read a dynamically sized slize from the given cursor
+pub fn read_slice<'de>(slice: &'de [u8], index: &mut usize) -> Option<&'de [u8]> {
+    let length = read_variable_length_integer(slice, index)?;
```

You can cast it to usize right away to avoid multiple casts below

---

i also using `try_into()`, i thought about 32bits embedded, there a cast could silently fail